.github/workflows/test_on_push.yml

```diff
@@ -45,12 +45,10 @@ jobs:
         python-version: ["3.8", "3.9", "3.10", "3.11"]
 
     steps:
-      - uses: actions/checkout@v3
-      - id: setup-python
-        name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+      - name: Check out PyBaMM repository
+        uses: actions/checkout@v3
         with:
-          python-version: ${{ matrix.python-version }}
+          fetch-depth: 0
```

Comment:
Nitpicking here, but is this required? This would fetch the entire git history, slowing down this step maybe by a few seconds. Is the cache stored using git?

Reply:
Not really, was just experimenting – I forgot to remove this 😄

Reply:
We can try getting rid of it!

Reply:
Done!